### ISSUE TITLE
Change In Blank Constraint Validation Between Grails 3.1.9 And Grails 3.3.1

### ISSUE THREAD

**Issue by aaron-brown** [grails/grails-core#10846]
While working on a migration of an application from Grails 3.1.9 to Grails 3.3.1, I cam across some inconsistent behavior with regard to validating the ```blank: false``` Constraint:

```groovy
class Foo {

    String fooField

    static constraints = {
        fooField(nullable: false, blank: false)
    }
}
```

In particular, what I've found is that there is an inoconsistency specifically revolving around validation of the following:

```groovy
Foo foo = new Foo()
foo.fooField = ' '
assert foo.validate() == false
```

This case will pass in Grails 3.1.9, but fail in Grails 3.3.1.

Using a debugger to trace through each version, it looks like in Grails 3.1.9 [**org.grails.validaiton.BlankConstraint**](https://github.com/grails/grails-core/blob/v3.3.1/grails-validation/src/main/groovy/org/grails/validation/BlankConstraint.java) drives the validation, which utilizes the [```isBlank()``` method of **GrailsStringUtils**](https://github.com/grails/grails-core/blob/v3.3.1/grails-core/src/main/groovy/grails/util/GrailsStringUtils.groovy#L167-L169), which applies a ```trim()```.

In contrast, Grails 3.3.1 uses [**org.grails.datastore.gorm.validation.constraints.BlankConstraint**](https://github.com/grails/grails-data-mapping/blob/v6.1.8/grails-datastore-gorm-validation/src/main/groovy/org/grails/datastore/gorm/validation/constraints/BlankConstraint.java) to drive Constraint Validation, which uses the [**org.springframework.util.StringUtils** ```isEmpty()``` method](https://github.com/spring-projects/spring-framework/blob/v5.0.1.RELEASE/spring-core/src/main/java/org/springframework/util/StringUtils.java#L88-L90). Note that this _does not_ apply a ```trim()``` implicitly to the String ```' '```, and therefore this execution returns the opposite equality to the **GrailsStringUtils** method.

This has the affect of creating an inconsistency in validation behavior.

I have provided sample Grails Projects in [3.1.9](https://github.com/aaron-brown/grails-validation-testing-3-1-9) and [3.3.1](https://github.com/aaron-brown/grails-validation-testing-3-3-1) to demonstrate this behavior both in the ```check``` test suite, as well as via a **TestController**.

Both applications have the same **Foo** Domain and **Bar** Validateable Objects, Tests, and **TestController**; the only difference is the Grails Version.

I am not sure between what two versions of Grails this behavior more accurrately changed, but it at least occurred between 3.1.9 and 3.3.1.

As an aside, the following method form appears to validate correctly:

```groovy
Foo foo = new Foo(fooField: ' ')
assert foo.validate() == false
```

However, this is violating the ```nullable`` Constraint, as demonstrated in the **FooSpec**, and I believe this is as documented for the [data-binding behavior](https://docs.grails.org/latest/ref/Constraints/nullable.html):

> Web requests resulting from form submissions will have blank strings, not null, for input fields that have no value. Keep this in mind when doing mass property binding to properties that are not nullable. The default behavior is such that a blank string will not validate for nullable: false since the data binder will convert blank strings to null. This includes empty strings and blank strings. A blank string is any string such that the trim() method returns an empty string. To turn off the conversion of empty strings to null set the grails.databinding.convertEmptyStringsToNull property to false in application.groovy. See the data binding section for more details on data binding.

**Comment by jeffbrown** [comment]
> In particular, what I've found is that there is an inoconsistency specifically revolving around validation of the following:
> 
> Foo foo = new Foo()
> foo.fooField = '  '
> assert foo.validate() == false
> This case will pass in Grails 3.1.9, but fail in Grails 3.3.1.

I would expect `validate()` to return `false` since `blank: false` is set.

**Comment by aaron-brown** [comment]
@jeffbrown: What is the reason for using the Spring **StringUtils** over the Apache Commons **StringUtils**?

JavaDoc in Spring **StringUtils** even [says](https://github.com/spring-projects/spring-framework/blob/v5.0.1.RELEASE/spring-core/src/main/java/org/springframework/util/StringUtils.java#L41-L43):

> Mainly for internal use within the framework; consider<a href="http://commons.apache.org/proper/commons-lang/">Apache's Commons Lang</a> for a more comprehensive suite of {@code String} utilities

**Comment by jameskleeh** [comment]
@aaron-brown I would guess because the spring version was already on the classpath and the apache version was not.

**Comment by jeffbrown** [comment]
BTW... The reason that `Foo foo = new Foo(fooField: ' ')` validates correctly if `Foo` is a domain class is because that doesn't use Groovy's default property assignment behavior but instead invokes the Grails data binder which by default trims strings (`grails.databinding.trimStrings`) and converts empty strings to null (`grails.databinding.convertEmptyStringsToNull`).

**Comment by jeffbrown** [comment]
Fixed in GORM 6.1.9.
